query-engine/js-connectors/js/neon-js-connector/src/neon.ts

```diff
@@ -18,22 +18,31 @@ const TRANSACTION_ROLLBACK = 'ROLLBACK'
 class PrismaNeon implements Connector, Closeable {
   readonly flavor = 'postgres'
 
-  private client: Client
+  private client?: Client
+  private httpClient?: NeonQueryFunction<false, true>
```

Comment:
I see this "magic booleans" corresponds to:

```typescript
type ARRAY_MODE = false
type FULL_RESULTS = true
```

Could you please add these types outside the `class PrismaNeon` definition, for the sake of having explicit (type) settings?

Then, `private httpClient?: NeonQueryFunction<false, true>` would become `private httpClient?: NeonQueryFunction<ARRAY_MODE, FULL_RESULTS>`.

Reply:
sounds good, and also we can make it an enum in the next release